Demo/Demo-iOS/ViewController.swift

```diff
@@ -40,7 +39,7 @@ class ViewController: UIViewController {
   }
 
   func animate() {
-    imageView.animate(withGIFNamed: currentGIFName, animationBlock:  {
+    imageView.animate(withGIFNamed: currentGIFName, preparationBlock:  {
```

Comment:
`animationBlock` is never run because the GIF loops indefinitely. `animationBlock` is only called once the whole animation has finished. With `preparationBlock` it works as intended.

Reply:
Nice catch!